### ISSUE TITLE
Gothic I (DX9 Renderer) - Exception on Startup

### ISSUE THREAD

**Issue by anchorlightforge** [NVIDIAGameWorks/rtx-remix#151]
**Regression from Portal with RTX**
In order to hook Remix, this was tested using the mods "SystemPack" and "Legacy AltRenderer;" mod instructions as provided by a community member can be found here:  [Remix Compatibility Trello card](https://trello.com/c/cx2YOBZ3/44-gothic) / [Remix Discord Project thread](https://discord.com/channels/1028444667789967381/1100062088828829776/1100062088828829776)

- Tested with Remix 0.2.0
- RTX 3080
- Nvidia Studio Driver 531.41
- GOG - Modified (see Trello and Discord project thread)

**Attach logs!**
[NvRemixBridge.log](https://github.com/NVIDIAGameWorks/rtx-remix/files/11467196/NvRemixBridge.log)
[d3d9.log](https://github.com/NVIDIAGameWorks/rtx-remix/files/11467199/d3d9.log)
[GothicMod_d3d9.log](https://github.com/NVIDIAGameWorks/rtx-remix/files/11467204/GothicMod_d3d9.log)

**To Reproduce**
1. Open <pathToGame>\system\GothicStarter_mod.exe
2. Select "Play the Game" and launch with DirectX9, no anti-aliasing
3. Allow the splash screen to display
4. Exception displays before loading completes

**Expected behavior**
- Portal with RTX: Remix hooks and path-traces, but all HUD elements on menus and movies flash black and white; press Enter multiple times to skip to gameplay, where the HUD will be entirely invisible.  Marking UI elements in Remix will result in the screen no longer updating during gameplay.
- Remix 0.1.1 and most DXVK-Remix commits: Gothic will launch to a mostly black screen, with no gameplay output.

**Screenshots**
![GothicCrashRemix020](https://github.com/NVIDIAGameWorks/rtx-remix/assets/55258193/4dfebd15-f57e-4c22-8b05-82bc58de2b38)

**Comment by AlexDunn** [comment]
It would be worth retrying this game with the latest build.  Have you had better luck with remix-0.2.0 or the latest binaries from github actions?

**Comment by anchorlightforge** [comment]
This bug was originally introduced by 0.2.0, but new commits to DXVK haven't changed anything so far.  I haven't tested the latest bridge commits yet.

**Comment by kcrg** [comment]
I can see the same problem with Open Beta version

**Comment by NV-LL** [comment]
Hello! Will you please test this on the latest release (0.5) and let us know if the issue persists? Thank you!

**Comment by anchorlightforge** [comment]
After initial tests it does appear that this issue was fixed.  For documentation purposes I will include some information that made its way to the Remix Discord thanks to user sunnlok's testing c. February of this year:

> Got it working.  ...
https://github.com/SaiyansKing/Gothic-LegacyAltRenderer/blob/main/LegacyAltRenderer/D3D9Renderer.cpp#L249
this line needs to be initialized to nullptr to avoid the crash
```IDirect3DBaseTexture9* oldTexture{nullptr};```

With Remix 0.5.0, I tested the release build of the Legacy Alt Renderer without the above fix, and the game was still able to function.  I'll also note that compatibility has greatly improved with working HUD support now.  Very nice to see this title make progress as this game unfortunately regressed with the open source versions quite a bit until more recently.
